app/models/user.rb

```diff
@@ -0,0 +1,47 @@
+require 'bcrypt'
+
+class User
+
+include DataMapper::Resource
+
+  attr_accessor :password_confirmation
+  attr_accessor :password
+  attr_accessor :password_token
+  attr_accessor :password_token_timestamp
+
+  validates_confirmation_of :password
+  validates_uniqueness_of :email
+  validates_uniqueness_of :username
+  validates_presence_of :password
+  validates_length_of :username, min: 8
+  validates_length_of :password, min: 8
+  validates_format_of :email, as: :email_address
+
+  property :user_id, Serial
+  property :username, Text, required: true
+  property :email, String, required: true
+  property :password_digest, Text
+  property :password_token, String
+
+  has n, :peeps, through: Resource
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(username,password)
+    user = first(username: username)
+    if user && BCrypt::Password.new(user.password_digest) == password
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)